akka-docs/src/main/paradox/stream/operators/Source/maybe.md

```diff
@@ -14,9 +14,35 @@ Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if comp
 
 ## Description
 
-Materialize a @scala[`Promise[Option[T]]`] @java[`CompletionStage`] that if completed with a @scala[`Some[T]`] @java[`Optional`] 
-will emit that *T* and then complete the stream, or if completed with @scala[`None`] @java[`empty Optional`] complete the stream right away.
+Materialize a @scala[`Promise[Option[T]]`] @java[`CompletableFuture`] which controls what element will be emitted by the
+Source.
 
+* If the materialized promise is completed with a @scala[`Some`]@java[non-empty `Optional`],
+  that value will be produced downstream, followed by completion.
+* If the materialized promise is completed with a @scala[`None`]@java[empty `Optional`],
+  no value will be produced downstream and completion will be signalled immediately.
+* If the materialized promise is completed with a failure, then the returned source will terminate with that error.
+* If the downstream of this source cancels before the promise has been completed, then the promise will be completed
+  with @scala[`None`]@java[empty `Optional`].
```

Comment:
Perhaps "cancels or fails" even if failure also means also a cancellation, just to make it explicit that we don't propagate downstream failure into the future.